pyomo/common/download.py

```diff
@@ -293,22 +294,27 @@ def destination(self):
 
     def retrieve_url(self, url):
         """Return the contents of a URL as an io.BytesIO object"""
+        ctx = ssl.create_default_context()
+        if self.cacert:
+            ctx.load_verify_locations(cafile=self.cacert)
+        if self.insecure:
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
         try:
-            ctx = ssl.create_default_context()
-            if self.cacert:
-                ctx.load_verify_locations(cafile=self.cacert)
-            if self.insecure:
-                ctx.check_hostname = False
-                ctx.verify_mode = ssl.CERT_NONE
             fetch = request.urlopen(url, context=ctx)
-        except:
+        except urllib_error.HTTPError as e:
+            if e.code != 403: # Forbidden
+                raise
+            fetch = None
+        if fetch is None:
```

Comment:
Can you explain the logic behind removing this from the `try-except` block? Wouldn't it be less messy to just keep it all within the `except`?

Reply:
If we keep it in the try-except, then the exception will be raised with the less-than helpful "while processing this exception, another exception occurred."  By moving it out, the second exception is "detangled" from the first.

Reply:
... Fine, but I want it noted that I don't like it from a readability perspective.